ingest-app/src/cmr/ingest/services/granule_bulk_update_service.clj

```diff
@@ -1,79 +1,162 @@
+                    (error "validate-and-save-bulk-granule-update caught exception:" e)
+                    (let [message (.getMessage e)
+                          user-facing-message (if (string/includes? message "GBUT_PN_I")
+                                                "Granule bulk update task name needs to be unique within the provider."
+                                                message)]
+                      (errors/throw-service-errors
+                       :invalid-data
+                       [(str "error creating granule bulk update task: " user-facing-message)]))))]
+    ;; Queue the granules bulk update event
+    (ingest-events/publish-gran-bulk-update-event
+     context
+     (ingest-events/granules-bulk-event provider-id task-id user-id instructions))
+    task-id))
+
+(defn handle-granules-bulk-event
+  "For each granule-ur, queueu granule bulk update messages"
+  [context provider-id task-id bulk-update-params user-id]
+  (doseq [instruction bulk-update-params]
+    (ingest-events/publish-gran-bulk-update-event
+     context
+     (ingest-events/ingest-granule-bulk-update-event provider-id task-id user-id instruction))))
+
+(defmulti add-opendap-url
+  "Add OPeNDAP url to the given granule concept."
+  (fn [context concept url]
+    (mt/format-key (:format concept))))
+
+(defmethod add-opendap-url :echo10
+  [context concept url]
+  (echo10/add-opendap-url (:metadata concept) url))
+
+(defmethod add-opendap-url :default
+  [context concept url]
+  (errors/throw-service-errors
+   :invalid-data (format "Add OPeNDAP url is not supported for format [%s]" (:format concept))))
+
+(defmulti update-granule-concept
+  "Perform the update of the granule concept."
+  (fn [context concept bulk-update-params user-id]
+    (keyword (:event-type bulk-update-params))))
+
+(defmethod update-granule-concept :UPDATE_FIELD:OPeNDAPLink
+  [context concept bulk-update-params user-id]
+  (let [{:keys [format metadata]} concept
+        {:keys [granule-ur url]} bulk-update-params
+        updated-metadata (add-opendap-url context concept url)]
+    (if-let [err-messages (:errors updated-metadata)]
+      (errors/throw-service-errors :invalid-data err-messages)
+      (-> concept
+          (assoc :metadata updated-metadata)
+          (update :revision-id inc)
+          (assoc :revision-date (time-keeper/now))
+          (assoc :user-id user-id)))))
+
+(defmethod update-granule-concept :default
+[context concept bulk-update-params user-id]
+)
+
+(defn- update-granule-concept-and-status
+  "Perform update for the granule concept and granule bulk update status."
+  [context task-id concept granule-ur bulk-update-params user-id]
+  (if-let [updated-concept (update-granule-concept context concept bulk-update-params user-id)]
+    (do
+      (ingest-service/save-granule context updated-concept)
+      (data-granule-bulk-update/update-bulk-update-task-granule-status
+       context task-id granule-ur bulk-update-service/updated-status ""))
```

Comment:
Why pass an empty string here?

Reply:
I don't want to introduce a new function signature for empty status message.